psydac/linalg/stencil.py

```diff
@@ -1410,6 +1414,9 @@ def diagonal(self, *, inverse = False, out = None):
         diag = self._data[diagonal_indices]
         data = out._data if out else None
 
+        if sqrt:
+            diag = np.sqrt(diag)
+
```

Comment:
This creates a new NumPy array with its own storage, which should not be necessary at this stage. Instead, you could pass the `out` argument to `np.sqrt` and move this code to **after** the next `if/elif/else` block:
```python
if sqrt:
    data = np.sqrt(data, out=data)
```

Reply:
Good point, is it not enough to write `np.sqrt(data, out=data)` (and `np.divide(1, diag, out=data)`) for updating data in place?

Reply:
Yes, the assignment is just syntactic sugar.